### ISSUE TITLE
setting --strict option always return RC 2

### ISSUE THREAD

**Issue by GwendolineFct** [ansible/ansible-lint#2592]
##### Summary

Whenever --strict is provided the return code will always be 2 regardless of errors or not

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible [core 2.12.9]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/gitlab/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.9/dist-packages/ansible
  ansible collection location = /home/gitlab/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110]
  jinja version = 3.1.2
  libyaml = True

ansible-lint --version
ansible-lint 6.8.1 using ansible 2.12.9
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```console
/usr/local/bin/ansible-lint --strict --offline --profile=production --skip-list=name[casing],name[template],schema[meta],no-handler,template-instead-of-copy tests/playbook-test-ansible-master-install.yml ; echo "return code = $?"

Passed with production profile: 0 failure(s), 0 warning(s) on 11 files.
return code = 2
```

looking at the code in [src/ansiblelint/app.py](https://github.com/ansible/ansible-lint/blob/v6.8.1/src/ansiblelint/app.py#L200) makes it obvious why it fails

```python
if not self.options.strict and (mark_as_success or not summary.failures):
    return SUCCESS_RC
return VIOLATIONS_FOUND_RC
```

##### Desired Behavior

setting --strict should only return RC 2 when any errors or warning detected

ie 

setting --strict should return RC 0 when no errors nor warning detected

```python
if not self.options.strict and (mark_as_success or not summary.failures):
    return SUCCESS_RC
if self.options.strict and not summary.failures and not summary.warnings:
    return SUCCESS_RC
return VIOLATIONS_FOUND_RC
```

##### Actual Behavior

setting --strict always return RC 2 even when no errors nor warning are detected
